src/main/java/uk/gov/ida/verifyserviceprovider/services/NonMatchingAssertionService.java

```diff
@@ -73,10 +75,22 @@ public TranslatedNonMatchingResponseBody translateSuccessResponse(List<Assertion
         validate(authnAssertion, mdsAssertion, expectedInResponseTo, expectedLevelOfAssurance, levelOfAssurance);
 
         String nameID = mdsAssertion.getSubject().getNameID().getValue();
+        String issuerID = mdsAssertion.getIssuer().getValue();
+
+        String authnContextClassRef =  extractLevelOfAssuranceStringFrom(authnAssertion);
+
+        String hashId = userIdHashFactory.hashId(
+                issuerID,
+                nameID,
+                Arrays.stream(uk.gov.ida.saml.core.domain.AuthnContext.values())
```

Comment:
Sorry to quibble over style but I would recommend using an import here rather than fully qualifying the class name, and I also think this section of code is complex enough that it's would be worth pulling it out into a local variable, rather than doing it all inside the method call brackets.  Obviously this isn't crucial to change though.